python/graphstorm/gconstruct/transform.py

```diff
@@ -255,12 +282,152 @@ def process_labels(data, label_confs, is_node):
+        return {train_mask_name: train_mask,
+                val_mask_name: val_mask,
+                test_mask_name: test_mask}
+
+class ClassificationProcessor(LabelProcessor):
+    """ Process the label for the classification task.
+    """
+
+    def __call__(self, data):
+        """ Process the label for classification.
+
+        This performs data split on the nodes/edges and generates training/validation/test masks.
+
+        Parameters
+        ----------
+        data : dict of Tensors
+            All data associated with nodes/edges of a node/edge type.
+
+        Returns
+        -------
+        dict of Tensors : it contains the labels as well as training/val/test splits.
+        """
+        assert self.col_name in data, f"The label column {self.col_name} does not exist."
+        label = data[self.col_name]
+        assert np.issubdtype(label.dtype, np.integer) \
+                or np.issubdtype(label.dtype, np.floating), \
+                "The labels for classification have to be integers."
+        valid_label_idx = get_valid_label_index(label)
+        def permute_idx():
+            return np.random.permutation(valid_label_idx)
+        res = self.data_split(permute_idx, len(label))
+        res[self.label_name] = np.int32(label)
+        return res
+
+class RegressionProcessor(LabelProcessor):
+    """ Process the label for the regression task.
+    """
+
+    def __call__(self, data):
+        """ Process the label for regression.
+
+        This performs data split on the nodes/edges and generates training/validation/test masks.
+
+        Parameters
+        ----------
+        data : dict of Tensors
+            All data associated with nodes/edges of a node/edge type.
+
+        Returns
+        -------
+        dict of Tensors : it contains the labels as well as training/val/test splits.
+        """
+        assert self.col_name in data, f"The label column {self.col_name} does not exist."
+        label = data[self.col_name]
+        valid_label_idx = get_valid_label_index(label)
+        def permute_idx():
+            return np.random.permutation(valid_label_idx)
+        res = self.data_split(permute_idx, len(label))
+        res[self.label_name] = label
+        return res
+
+class LinkPredictionProcessor(LabelProcessor):
+    """ Process the label for the link prediction task.
+    """
+
+    def __call__(self, data):
+        """ Process the label for link prediction.
+
+        This performs data split on the edges and generates training/validation/test masks.
+
+        Parameters
+        ----------
+        data : dict of Tensors
+            All data associated with nodes/edges of a node/edge type.
+
+        Returns
+        -------
+        dict of Tensors : it contains training/val/test splits for link prediction.
+        """
+        # Any column in the data can define the number of samples in the data.
+        assert len(data) > 0, "The edge data is empty."
+        for val in data.values():
```

Comment:
Why not just num_samples = len(data[0])

Reply:
data is a dict. we cannot use `[0]` to index it.